File: android/src/main/java/kr/co/wisetracker/WiseTrackerManager.java
package kr.co.wisetracker;

import android.util.Log;
import android.widget.Toast;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;

import javax.annotation.Nonnull;

import kr.co.wisetracker.tracker.WiseTracker;

public class WiseTrackerManager extends ReactContextBaseJavaModule {

    private final String TAG = "WiseTrackerManager";

    public WiseTrackerManager(@Nonnull ReactApplicationContext reactContext) {
        super(reactContext);
    }

    @Nonnull
    @Override
    public String getName() {
        return "WiseTrackerManager";
    }

    @ReactMethod
    public void initialization() {
        try {
            Toast.makeText(getReactApplicationContext(), "init", Toast.LENGTH_SHORT).show();
            WiseTracker.init(getReactApplicationContext());
        } catch (Exception e) {
            Log.e(TAG, "initialization error !!", e);
        }
    }

    @ReactMethod
    public void initStart() {
        try {
            WiseTracker.initStart(getReactApplicationContext());
        } catch (Exception e) {
            Log.e(TAG, "initStart error !!", e);
        }
    }

    @ReactMethod
    public void initEnd() {
        try {
            WiseTracker.initEnd();
        } catch (Exception e) {
            Log.e(TAG, "initEnd error !!", e);
        }
    }


    @ReactMethod
    public void sendTransaction() {
        try {
            WiseTracker.sendTransaction();
        } catch (Exception e) {
            Log.e(TAG, "sendTransaction error !!", e);
        }
    }

    @ReactMethod
    public void initPushSet(String key, String value) {
        try {
            WiseTracker.initPushSet(key, value);
        } catch (Exception e) {
            Log.e(TAG, "initPushSet error !!", e);
        }
    }

    @ReactMethod
    public void putInitData(String key, String value) {
        try {
            WiseTracker.putInitData(key, value);
        } catch (Exception e) {
            Log.e(TAG, "putInitData error !!", e);
        }
    }

    @ReactMethod
    public void startPage(String data) {
        try {
            WiseTracker.startPage(data);
        } catch (Exception e) {
            Log.e(TAG, "startPage error !!", e);
        }
    }

    @ReactMethod
    public void endPage(String data) {
        try {
            WiseTracker.endPage(data);
        } catch (Exception e) {
            Log.e(TAG, "endPage error !!", e);
        }
    }

    @ReactMethod
    public void endStartPage(String data) {
        try {
            WiseTracker.endStartPage(data);
        } catch (Exception e) {
            Log.e(TAG, "endStartPage error !!", e);
        }
    }

    @ReactMethod
    public void putSessionData(String key, String value) {
        try {
            WiseTracker.putSessionData(key, value);
        } catch (Exception e) {
            Log.e(TAG, "putSessionData error !!", e);
        }
    }

    @ReactMethod
    public void putSessionReferrer(String referrer) {
        try {
            WiseTracker.putSessionReferrer(referrer);
        } catch (Exception e) {
            Log.e(TAG, "putSessionReferrer error !!", e);
        }
    }

    @ReactMethod
    public void putInstallReferrer(String referrer) {
        try {
            WiseTracker.putInstallReferrer(referrer);
        } catch (Exception e) {
            Log.e(TAG, "putSessionReferrer error !!", e);
        }
    }

    @ReactMethod
    public void sendGoalData() {
        try {
            WiseTracker.sendGoalData();
        } catch (Exception e) {
            Log.e(TAG, "setGoal error !!", e);
        }
    }

    @ReactMethod
    public void setGoal(String key, String value) {
        try {
            WiseTracker.setGoal(key, Double.valueOf(value));
        } catch (Exception e) {
            Log.e(TAG, "setGoal error !!", e);
        }
    }

    @ReactMethod
    public void setAcceptPushReceived(String isAccepted) {
        try {
            WiseTracker.setAcceptPushReceived(Boolean.valueOf(isAccepted));
        } catch (Exception e) {
            Log.e(TAG, "setAcceptPushReceived error !!", e);
        }
    }

    @ReactMethod
    public void setGoalCustomMvtTag(String key, String value) {
        try {
            WiseTracker.setGoalCustomMvtTag(key, value);
        } catch (Exception e) {
            Log.e(TAG, "setGoalCustomMvtTag error !!", e);
        }
    }

    @ReactMethod
    public void setGoalProduct(String code) {
        try {
            WiseTracker.setGoalProduct(code);
        } catch (Exception e) {
            Log.e(TAG, "setGoalProduct error !!", e);
        }
    }

    @ReactMethod
    public void setGoalProductArray(String[] array) {
        try {
            WiseTracker.setGoalProductArray(array);
        } catch (Exception e) {
            Log.e(TAG, "setGoalProductArray error !!", e);
        }
    }

    @ReactMethod
    public void setGoalProductType(String type) {
        try {
            WiseTracker.setGoalProductType(type);
        } catch (Exception e) {
            Log.e(TAG, "setGoalProductType error !!", e);
        }
    }

    @ReactMethod
    public void setGoalProductTypeArray(String[] array) {
        try {
            WiseTracker.setGoalProductTypeArray(array);
        } catch (Exception e) {
            Log.e(TAG, "setGoalProductTypeArray error !!", e);
        }
    }

    @ReactMethod
    public void setGoalProductCategory(String code) {
        try {
            WiseTracker.setGoalProductCategory(code);
        } catch (Exception e) {
            Log.e(TAG, "setGoalProductCategory error !!", e);
        }
    }

    @ReactMethod
    public void setExhibit(String exhibit) {
        try {
            WiseTracker.setExhibit(exhibit);
        } catch (Exception e) {
            Log.e(TAG, "setExhibit error !!", e);
        }
    }

    @ReactMethod
    public void setProduct(String key, String value) {
        try {
            WiseTracker.setProduct(key, value);
        } catch (Exception e) {
            Log.e(TAG, "setProduct error !!", e);
        }
    }

    @ReactMethod
    public void setProductType(String type) {
        try {
            WiseTracker.setProductType(type);
        } catch (Exception e) {
            Log.e(TAG, "setProductType error !!", e);
        }
    }

    @ReactMethod
    public void setProductCategory(String category, String name) {
        try {
            WiseTracker.setProductCategory(category, name);
        } catch (Exception e) {
            Log.e(TAG, "setProductCategory error !!", e);
        }
    }

    @ReactMethod
    public void setOrderCustomMvtTag(String key, String value) {
        try {
            WiseTracker.setOrderCustomMvtTag(key, value);
        } catch (Exception e) {
            Log.e(TAG, "setOrderCustomMvtTag error !!", e);
        }
    }

    @ReactMethod
    public void setOrderCustomMvtTagArray(String key, String[] array) {
        try {
            WiseTracker.setOrderCustomMvtTagArray(key, array);
        } catch (Exception e) {
            Log.e(TAG, "setOrderCustomMvtTagArray error !!", e);
        }
    }

    @ReactMethod
    public void setOrderProduct(String code) {
        try {
            WiseTracker.setOrderProduct(code);
        } catch (Exception e) {
            Log.e(TAG, "setOrderProduct error !!", e);
        }
    }

    @ReactMethod
    public void setOrderProductArray(String[] array) {
        try {
            WiseTracker.setOrderProductArray(array);
        } catch (Exception e) {
            Log.e(TAG, "setOrderProductArray error !!", e);
        }
    }

    @ReactMethod
    public void setOrderProductType(String type) {
        try {
            WiseTracker.setOrderProductType(type);
        } catch (Exception e) {
            Log.e(TAG, "setOrderProductType error !!", e);
        }
    }

    @ReactMethod
    public void setOrderProductTypeArray(String[] array) {
        try {
            WiseTracker.setOrderProductTypeArray(array);
        } catch (Exception e) {
            Log.e(TAG, "setOrderProductTypeArray error !!", e);
        }
    }

    @ReactMethod
    public void setOrderProductCategory(String category) {
        try {
            WiseTracker.setOrderProductCategory(category);
        } catch (Exception e) {
            Log.e(TAG, "setOrderProductCategory error !!", e);
        }
    }

    @ReactMethod
    public void setOrderProductCategoryArray(String[] array) {
        try {
            WiseTracker.setOrderProductCategoryArray(array);
        } catch (Exception e) {
            Log.e(TAG, "setOrderProductCategoryArray error !!", e);
        }
    }

    @ReactMethod
    public void setOrderAmount(String value) {
        try {
            WiseTracker.setOrderAmount(Double.valueOf(value));
        } catch (Exception e) {
            Log.e(TAG, "setOrderAmount error !!", e);
        }
    }

    @ReactMethod
    public void setOrderAmountArray(String[] array) {
        try {
            if (array == null || array.length == 0) {
                return;
            }
            double[] valueArray = new double[array.length];
            for (int i = 0; i < array.length; i++) {
                valueArray[i] = Double.valueOf(array[i]);
                Log.d("cordova", "value[" + i + "] = " + valueArray[i]);
            }
            WiseTracker.setOrderAmountArray(valueArray);
        } catch (Exception e) {
            Log.e(TAG, "setOrderAmountArray error !!", e);
        }
    }

    @ReactMethod
    public void setOrderConversionData(String key, String value) {
        try {
            WiseTracker.setOrderConversionData(key, Double.valueOf(value));
        } catch (Exception e) {
            Log.e(TAG, "setOrderConversionData error !!", e);
        }
    }

    @ReactMethod
    public void setOrderConversionDataArray(String key, String[] array) {
        try {
            if (array == null || array.length == 0) {
                return;
            }
            double[] valueArray = new double[array.length];
            for (int i = 0; i < array.length; i++) {
                valueArray[i] = Double.valueOf(array[i]);
                Log.d("cordova", "value[" + i + "] = " + valueArray[i]);
            }
            WiseTracker.setOrderConversionDataArray(key, valueArray);
        } catch (Exception e) {
            Log.e(TAG, "setOrderConversionDataArray error !!", e);
        }
    }

    @ReactMethod
    public void useIkwdWithConversion(String value) {
        try {
            WiseTracker.useIkwdWithConversion(value);
        } catch (Exception e) {
            Log.e(TAG, "useIkwdWithConversion error !!", e);
        }
    }

    @ReactMethod
    public void useMvt1WithConversion(String value) {
        try {
            WiseTracker.useMvt1WithConversion(value);
        } catch (Exception e) {
            Log.e(TAG, "useMvt1WithConversion error !!", e);
        }
    }

    @ReactMethod
    public void useMvt2WithConversion(String value) {
        try {
            WiseTracker.useMvt2WithConversion(value);
        } catch (Exception e) {
            Log.e(TAG, "useMvt2WithConversion error !!", e);
        }
    }

    @ReactMethod
    public void useMvt3WithConversion(String value) {
        try {
            WiseTracker.useMvt3WithConversion(value);
        } catch (Exception e) {
            Log.e(TAG, "useMvt3WithConversion error !!", e);
        }
    }

    @ReactMethod
    public void setOrderNo(String orderNo) {
        try {
            WiseTracker.setOrderNo(orderNo);
        } catch (Exception e) {
            Log.e(TAG, "setOrderNo error !!", e);
        }
    }

    @ReactMethod
    public void setOrderQuantity(String value) {
        try {
            WiseTracker.setOrderQuantity(Integer.valueOf(value));
        } catch (Exception e) {
            Log.e(TAG, "setOrderQuantity error !!", e);
        }
    }

    @ReactMethod
    public void setOrderQuantityArray(String[] array) {
        try {
            if (array == null || array.length == 0) {
                return;
            }
            int[] valueArray = new int[array.length];
            for (int i = 0; i < array.length; i++) {
                valueArray[i] = Integer.valueOf(array[i]);
            }
            WiseTracker.setOrderQuantityArray(valueArray);
        } catch (Exception e) {
            Log.e(TAG, "setOrderQuantityArray error !!", e);
        }
    }

    @ReactMethod
    public void setContents(String contents) {
        try {
            WiseTracker.setContents(contents);
        } catch (Exception e) {
            Log.e(TAG, "setContents error !!", e);
        }
    }

    @ReactMethod
    public void setPageIdentity(String identify) {
        try {
            WiseTracker.setPageIdentity(identify);
        } catch (Exception e) {
            Log.e(TAG, "setPageIdentity error !!", e);
        }
    }

    @ReactMethod
    public void setSearchKeyword(String keyword) {
        try {
            WiseTracker.setSearchKeyword(keyword);
        } catch (Exception e) {
            Log.e(TAG, "setSearchKeyword error !!", e);
        }
    }

    @ReactMethod
    public void setGoalSearchKeyword(String goalSearchKeyword) {
        try {
            WiseTracker.setGoalSearchKeyword(goalSearchKeyword);
        } catch (Exception e) {
            Log.e(TAG, "setGoalSearchKeyword error !!", e);
        }
    }

    @ReactMethod
    public void setOrderSearchKeyword(String orderSearchKeyword) {
        try {
            WiseTracker.setOrderSearchKeyword(orderSearchKeyword);
        } catch (Exception e) {
            Log.e(TAG, "setOrderSearchKeyword error !!", e);
        }
    }

    @ReactMethod
    public void setOrderSearchKeywordArray(String[] array) {
        try {
            WiseTracker.setOrderSearchKeywordArray(array);
        } catch (Exception e) {
            Log.e(TAG, "setOrderSearchKeywordArray error !!", e);
        }
    }

    @ReactMethod
    public void setSearchKeywordResult(String count) {
        try {
            WiseTracker.setSearchKeywordResult(Integer.valueOf(count));
        } catch (Exception e) {
            Log.e(TAG, "setSearchKeywordResult error !!", e);
        }
    }

    @ReactMethod
    public void setSearchKeywordCategory(String category) {
        try {
            WiseTracker.setSearchKeywordCategory(category);
        } catch (Exception e) {
            Log.e(TAG, "setSearchKeywordCategory error !!", e);
        }
    }

    @ReactMethod
    public void setMember(String member) {
        try {
            WiseTracker.setMember(member);
        } catch (Exception e) {
            Log.e(TAG, "setMember error !!", e);
        }
    }

    @ReactMethod
    public void setGender(String gender) {
        try {
            WiseTracker.setGender(gender);
        } catch (Exception e) {
            Log.e(TAG, "setGender error !!", e);
        }
    }

    @ReactMethod
    public void setAge(String age) {
        try {
            WiseTracker.setAge(age);
        } catch (Exception e) {
            Log.e(TAG, "setAge error !!", e);
        }
    }

    @ReactMethod
    public void setUserAttribute(String key, String value) {
        try {
            WiseTracker.setUserAttribute(key, value);
        } catch (Exception e) {
            Log.e(TAG, "setUserAttribute error !!", e);
        }
    }

    @ReactMethod
    public void setCustomMvtTag(String key, String value) {
        try {
            WiseTracker.setCustomMvtTag(key, value);
        } catch (Exception e) {
            Log.e(TAG, "setCustomMvtTag error !!", e);
        }
    }

    @ReactMethod
    public void setOrderDate(String orderDate) {
        try {
            WiseTracker.setOrderDate(orderDate);
        } catch (Exception e) {
            Log.e(TAG, "setOrderDate error !!", e);
        }
    }

    @ReactMethod
    public void setOrderDateArray(String orderDate, String count) {
        try {
            WiseTracker.setOrderDateArray(orderDate, Integer.valueOf(count));
        } catch (Exception e) {
            Log.e(TAG, "setOrderDateArray error !!", e);
        }
    }

    @ReactMethod
    public void setWisetrackerAppkey(String appKey) {
        try {
            WiseTracker.setWisetrackerAppkey(appKey);
        } catch (Exception e) {
            Log.e(TAG, "setWisetrackerAppkey error !!", e);
        }
    }

    @ReactMethod
    public void setWisetrackerDebugMode(String mode) {
        try {
            WiseTracker.setWisetrackerDebugMode(Boolean.valueOf(mode));
        } catch (Exception e) {
            Log.e(TAG, "setWisetrackerDebugMode error !!", e);
        }
    }

    @ReactMethod
    public void sendClickData(String key, String value) {
        try {
            WiseTracker.sendClickData(key, value);
        } catch (Exception e) {
            Log.e(TAG, "sendClickData error !!", e);
        }
    }

    @ReactMethod
    public void sendImmediatelyClickData(String key, String value) {
        try {
            WiseTracker.sendImmediatelyClickData(key, value);
        } catch (Exception e) {
            Log.e(TAG, "sendImmediatelyClickData error !!", e);
        }
    }

}

File: iOS/WiseTrackerManager.m
//
//  WiseTrackerManager.m
//  AwesomeProject
//
//  Created by Woncheol Heo on 2019. 4. 25..
//  Copyright © 2019년 Facebook. All rights reserved.
//

#import "WiseTrackerManager.h"
#import <React/RCTLog.h>
#import <WiseTracker/WiseTracker.h>

@implementation WiseTrackerManager

RCT_EXPORT_MODULE();

RCT_EXPORT_METHOD(setWisetrackerAppkey:(NSString *)appKey)
{
    [WiseTracker applicationKey:appKey];
    [WiseTracker setApplication: RCTSharedApplication()];
}

RCT_EXPORT_METHOD(initialization)
{
    dispatch_async(dispatch_get_main_queue(), ^{
        [WiseTracker initEnd];
    });
}

RCT_EXPORT_METHOD(startPage:(NSString *)pageId )
{
  [WiseTracker startPage:pageId];
}

RCT_EXPORT_METHOD(endPage:(NSString *)pageId)
{
  [WiseTracker endPage:pageId];
}

RCT_EXPORT_METHOD(sendTransactionImmediate)
{
  [WiseTracker sendTransactionImmediate];
}

RCT_EXPORT_METHOD(setGoal:(NSString *)key value:(NSInteger)value)
{
  RCTLogInfo(@"This func call [WiseTracker setGoal]!!!! awesome :%@ %ld", key, (long)value);
  [WiseTracker setGoal:key value:[NSNumber numberWithInteger:value]];
}

RCT_EXPORT_METHOD(endStartPage:(NSString *)endPageId key:(NSString *)startPageId)
{
  [WiseTracker endPage:endPageId];
  [WiseTracker startPage:startPageId];
}

RCT_EXPORT_METHOD(setGoalById:(NSString *)pageId key:(NSString *)key value:(NSInteger )value)
{
  [WiseTracker setGoal:key value:[NSNumber numberWithInteger:value] byId:pageId];
}

RCT_EXPORT_METHOD(setAcceptPushReceived:(BOOL)isAccept)
{
  [WiseTracker setAcceptPushReceived:isAccept];
}

RCT_EXPORT_METHOD(setAcceptPushReceivedById:(NSString *)pageId isAccept:(BOOL)isAccept)
{
  [WiseTracker setAcceptPushReceived:isAccept byId:pageId];
}

RCT_EXPORT_METHOD(setMember:(NSString *)isMember)
{
  [WiseTracker setMember:isMember];
}

RCT_EXPORT_METHOD(setGender:(NSString *)gender)
{
  [WiseTracker setGender:gender];
}

RCT_EXPORT_METHOD(setAge:(NSString *)age)
{
  [WiseTracker setAge:age];
}

RCT_EXPORT_METHOD(setUserAttribute:(NSString *)key value:(NSString *)value)
{
  [WiseTracker setUserAttribute:key attribute:value];
}

RCT_EXPORT_METHOD(setContents:(NSString *)contents)
{
  [WiseTracker setContents:contents];
}

RCT_EXPORT_METHOD(setContentsById:(NSString *)contents pageId:(NSString *)pageId)
{
  [WiseTracker setContents:contents byId:pageId];
}

RCT_EXPORT_METHOD(setPageIdentity:(NSString *)pi)
{
  [WiseTracker setPageIdentity:pi];
}

RCT_EXPORT_METHOD(setSearchKeyword:(NSString *)keyword)
{
  [WiseTracker setSearchKeyword:keyword];
}

RCT_EXPORT_METHOD(setGoalSearchKeyword:(NSString *)keyword)
{
  [WiseTracker setGoalSearchKeyword:keyword];
}

RCT_EXPORT_METHOD(setOrderSearchKeywordl:(NSString *)keyword)
{
  [WiseTracker setOrderSearchKeyword:keyword];
}

RCT_EXPORT_METHOD(setOrderSearchKeywordArray:(NSArray *)keywordArray)
{
  [WiseTracker setOrderSearchKeywordArray:keywordArray];
}

RCT_EXPORT_METHOD(setSearchKeywordById:(NSString *)keyword pageId:(NSString *)pageId)
{
  [WiseTracker setSearchKeyword:keyword byId:pageId];
}

RCT_EXPORT_METHOD(setSearchKeywordResult:(NSInteger)cnt)
{
  [WiseTracker setSearchKeywordResult:cnt];
}

RCT_EXPORT_METHOD(setSearchKeywordResultById:(NSInteger)cnt pageId:(NSString *)pageId)
{
  [WiseTracker setSearchKeywordResult:cnt byId:pageId];
}

RCT_EXPORT_METHOD(setSearchKeywordCategory:(NSString *)category)
{
  [WiseTracker setSearchKeywordCategory:category];
}

RCT_EXPORT_METHOD(setSearchKeywordCategoryById:(NSString *)category pageId:(NSString *)pageId)
{
  [WiseTracker setSearchKeywordCategory:category byId:pageId];
}

RCT_EXPORT_METHOD(setCustomMvtTag:(NSString *)key value:(NSString *)value)
{
  [WiseTracker setCustomMvtTag:key value:value];
}

RCT_EXPORT_METHOD(setCustomMvtTag:(NSString *)key value:(NSString *)value pageId:(NSString *)pageId)
{
  [WiseTracker setCustomMvtTag:key value:value byId:pageId];
}

RCT_EXPORT_METHOD(setGoalCustomMvtTag:(NSString *)key value:(NSString *)value)
{
  [WiseTracker setGoalCustomMvtTag:key value:value];
}

RCT_EXPORT_METHOD(setGoalCustomMvtTag:(NSString *)key value:(NSString *)value pageId:(NSString *)pageId)
{
  [WiseTracker setGoalCustomMvtTag:key value:value byId:pageId];
}

RCT_EXPORT_METHOD(setGoalCustomMvtTagById:(NSString *)key value:(NSString *)value pageId:(NSString *)pageId)
{
  [WiseTracker setGoalCustomMvtTag:key value:value byId:pageId];
}

RCT_EXPORT_METHOD(setOrderCustomMvtTag:(NSString *)key value:(NSString *)value)
{
  [WiseTracker setOrderCustomMvtTag:key value:value];
}

RCT_EXPORT_METHOD(setOrderCustomMvtTagArray:(NSString *)key mvtTagArray:(NSArray *)mvtTagArray)
{
  [WiseTracker setOrderCustomMvtTagArray:key value:mvtTagArray];
}

RCT_EXPORT_METHOD(setOrderCustomMvtTagById:(NSString *)key value:(NSString *)value pageId:(NSString *)pageId )
{
  [WiseTracker setOrderCustomMvtTag:key value:value byId:pageId];
}

RCT_EXPORT_METHOD(setExhibit:(NSString *)exhibit)
{
  [WiseTracker setExhibit:exhibit];
}

RCT_EXPORT_METHOD(setOrderNo:(NSString *)orderNo)
{
  [WiseTracker setOrderNo:orderNo];
}

RCT_EXPORT_METHOD(setOrderNoById:(NSString *)orderNo pageId:(NSString *)pageId)
{
  [WiseTracker setOrderNo:orderNo byId:pageId];
}


RCT_EXPORT_METHOD(setProduct:(NSString *)code name:(NSString *)name)
{
  [WiseTracker setProduct:code name:name];
}

RCT_EXPORT_METHOD(setProductById:(NSString *)code name:(NSString *)name pageId:(NSString *)pageId)
{
  [WiseTracker setProduct:code name:name byId:pageId];
}

RCT_EXPORT_METHOD(setGoalProduct:(NSString *)code)
{
  [WiseTracker setGoalProduct:code];
}

RCT_EXPORT_METHOD(setGoalProductById:(NSString *)code pageId:(NSString *)pageId)
{
  [WiseTracker setGoalProduct:code byId:pageId];
}

RCT_EXPORT_METHOD(setGoalProductArray:(NSArray *)productArray)
{
  [WiseTracker setGoalProductArray:productArray];
}

RCT_EXPORT_METHOD(setGoalProductArrayById:(NSArray *)productArray pageId:(NSString *)pageId)
{
  [WiseTracker setGoalProductArray:productArray byId:pageId];
}

RCT_EXPORT_METHOD(setOrderProduct:(NSString *)code)
{
  [WiseTracker setOrderProduct:code];
}

RCT_EXPORT_METHOD(setOrderProductById:(NSString *)code pageId:(NSString *)pageId)
{
  [WiseTracker setOrderProduct:code byId:pageId];
}

RCT_EXPORT_METHOD(setOrderProductArray:(NSArray *)productArray)
{
  [WiseTracker setOrderProductArray:productArray];
}

RCT_EXPORT_METHOD(setOrderProductArrayById:(NSArray *)productArray pageId:(NSString *)pageId)
{
  [WiseTracker setOrderProductArray:productArray byId:pageId];
}

RCT_EXPORT_METHOD(setProductCategory:(NSString *)code name:(NSString *)name )
{
  [WiseTracker setProductCategory:code name:name];
}

RCT_EXPORT_METHOD(setProductCategoryById:(NSString *)code name:(NSString *)name pageId:(NSString *)pageId)
{
  [WiseTracker setProductCategory:code name:name byId:pageId];
}

RCT_EXPORT_METHOD(setGoalProductCategory:(NSString *)code)
{
  [WiseTracker setGoalProductCategory:code];
}

RCT_EXPORT_METHOD(setGoalProductCategoryArray:(NSArray *)categoryArray)
{
  [WiseTracker setGoalProductCategoryArray:categoryArray];
}

RCT_EXPORT_METHOD(setGoalProductCategoryArrayById:(NSArray *)categoryArray pageId:(NSString *)pageId)
{
  [WiseTracker setGoalProductArray:categoryArray byId:pageId];
}

RCT_EXPORT_METHOD(setOrderProductCategory:(NSString *)code)
{
  [WiseTracker setOrderProductCategory:code];
}

RCT_EXPORT_METHOD(setOrderProductCategoryById:(NSString *)code pageId:(NSString *)pageId)
{
  [WiseTracker setOrderProductCategory:code byId:pageId];
}

RCT_EXPORT_METHOD(setOrderProductCategoryArray:(NSArray *)categoryArray)
{
  [WiseTracker setOrderProductCategoryArray:categoryArray];
}

RCT_EXPORT_METHOD(setOrderProductCategoryArrayById:(NSArray *)categoryArray pageId:(NSString *)pageId)
{
  [WiseTracker setOrderProductCategoryArray:categoryArray byId:pageId];
}

RCT_EXPORT_METHOD(setProductType:(NSString *)type)
{
  [WiseTracker setProductType:type];
}

RCT_EXPORT_METHOD(setProductTypeById:(NSString *)type pageId:(NSString *)pageId)
{
  [WiseTracker setProductType:type byId:pageId];
}

RCT_EXPORT_METHOD(setGoalProductType:(NSString *)type)
{
  [WiseTracker setGoalProductType:type];
}

RCT_EXPORT_METHOD(setGoalProductTypeById:(NSString *)type pageId:(NSString *)pageId)
{
  [WiseTracker setGoalProductType:type byId:pageId];
}

RCT_EXPORT_METHOD(setGoalProductTypeArray:(NSArray *)typeArray)
{
  [WiseTracker setGoalProductTypeArray:typeArray];
}

RCT_EXPORT_METHOD(setGoalProductTypeArrayById:(NSArray *)typeArray pageId:(NSString *)pageId)
{
  [WiseTracker setGoalProductTypeArray:typeArray byId:pageId];
}

RCT_EXPORT_METHOD(setOrderProductType:(NSString *)type)
{
  [WiseTracker setOrderProductType:type];
}

RCT_EXPORT_METHOD(setOrderProductTypeById:(NSString *)type pageId:(NSString *)pageId)
{
  [WiseTracker setOrderProductType:type byId:pageId];
}

RCT_EXPORT_METHOD(setOrderProductTypeArray:(NSArray *)typeArray)
{
  [WiseTracker setOrderProductTypeArray:typeArray];
}

RCT_EXPORT_METHOD(setOrderAmount:(NSNumber *)amount)
{
  [WiseTracker setOrderAmount:amount];
}

RCT_EXPORT_METHOD(setOrderAmountById:(NSInteger )amount pageId:(NSString *)pageId)
{
  [WiseTracker setOrderAmount:amount byId:pageId];
}

RCT_EXPORT_METHOD(setOrderAmountArray:(NSArray *)amountArray)
{
  [WiseTracker setOrderAmountArray:amountArray];
}

RCT_EXPORT_METHOD(setOrderAmountArrayById:(NSArray *)amountArray pageId:(NSString *)pageId)
{
  [WiseTracker setOrderAmountArray:amountArray byId:pageId];
}

RCT_EXPORT_METHOD(setOrderQuantity:(NSInteger)quantity)
{
  [WiseTracker setOrderQuantity:quantity];
}

RCT_EXPORT_METHOD(setOrderQuantityById:(NSInteger)quantity pageId:(NSString *)pageId)
{
  [WiseTracker setOrderQuantity:quantity byId:pageId];
}

RCT_EXPORT_METHOD(setOrderQuantityArray:(NSArray *)quantityArray)
{
  [WiseTracker setOrderQuantityArray:quantityArray];
}

RCT_EXPORT_METHOD(setOrderQuantityArrayById:(NSArray *)quantityArray pageId:(NSString *)pageId)
{
  [WiseTracker setOrderQuantityArray:quantityArray byId:pageId];
}

RCT_EXPORT_METHOD(sendClickData:(NSString *)eventCode eventName:(NSString *)eventName)
{
  [WiseTracker sendClickData:eventCode eventName:eventName];
}

RCT_EXPORT_METHOD(sendGoalData)
{
  [WiseTracker sendGoalData];
}

RCT_EXPORT_METHOD(setPaymentTypeForOllehTv:(NSString *)mCd sCd:(NSString *)sCd)
{
  [WiseTracker setPaymentTypeForOllehTv:mCd sCd:sCd];
}

RCT_EXPORT_METHOD(setOrderConversionData:(NSString *)key value:(NSNumber *)value)
{
  [WiseTracker setOrderConversionData:value key:key];
}

RCT_EXPORT_METHOD(setOrderConversionDataById:(NSString *)key value:(NSInteger)value pageId:(NSString *)pageId)
{
  [WiseTracker setOrderConversionData:value key:key byId:pageId];
}

RCT_EXPORT_METHOD(setOrderConversionDataArrayById:(NSArray *)dataArray key:(NSString *)key pageId:(NSString *)pageId)
{
  [WiseTracker setOrderConversionDataArray:dataArray key:key byId:pageId];
}

RCT_EXPORT_METHOD(setInternalEvent:(NSString *)value)
{
  [WiseTracker setInternalEvent:value];
}

RCT_EXPORT_METHOD(setOrderConversionDataArray:(NSArray *)dataArray key:(NSString *)key)
{
  [WiseTracker setOrderConversionDataArray:dataArray key:key];
}

RCT_EXPORT_METHOD(useIkwdWithConversion:(NSString *)value)
{
  [WiseTracker useIkwdWithConversion:value];
}

RCT_EXPORT_METHOD(useMvt1WithConversion:(NSString *)value)
{
  [WiseTracker useMvt1WithConversion:value];
}

RCT_EXPORT_METHOD(useMvt2WithConversion:(NSString *)value)
{
  [WiseTracker useMvt2WithConversion:value];
}

RCT_EXPORT_METHOD(useMvt3WithConversion:(NSString *)value)
{
  [WiseTracker useMvt3WithConversion:value];
}

RCT_EXPORT_METHOD(setOrderDate:(NSString *)date)
{
  [WiseTracker setOrderDate:date];
}

RCT_EXPORT_METHOD(setOrderDateArray:(NSString *)date itemCount:(NSInteger)itemCount)
{
  [WiseTracker setOrderDateArray:date itemCount:itemCount];
}

@end
